### PR TITLE
fix(satori): missing `status`, fix koishijs/koishi#1367, fix #241

### LINKED ISSUE #241: Bug: adapter-satori 连 Chronocat 后 READY 信令内没有 status 字段
见 https://github.com/koishijs/koishi/issues/1367

### PR DESCRIPTION
**Pull request by Lipraty**
#241 

koishijs/koishi#1367

### CODE REVIEW

**Comment by Lipraty** [comment]
? 怎么 CI 爆了

**Comment by MaikoTan** [comment]
> ? 怎么 CI 爆了

Looks like the `yakumo` should be updated, but overall the PR looks good to me.

File: adapters/satori/src/ws.ts [hunk 1]
```diff
@@ -58,6 +58,8 @@ export class SatoriAdapter<C extends Context = Context> extends Adapter.WsClient
     bot = new SatoriBot(this.ctx, login)
     bot.adapter = this
     bot.http = this.http
+    bot.status = Universal.Status.ONLINE // Will this happen?
+    setStatus(bot.status)
```
Comment (shigma) [inline adapters/satori/src/ws.ts:62]:
```suggestion
    bot.status = login.status
```

Comment (shigma) [reply adapters/satori/src/ws.ts:62]: 顺便可能需要把 this.bots.push 挪到上面，bot = 的下一行。

**Review by shigma — APPROVED** [review]

File: adapters/satori/src/ws.ts (hunk 1 above)
Comment (shigma) [reply adapters/satori/src/ws.ts:62]: 不小心点成 approve 了

**Comment by Lipraty** [comment]
Testing is completed and the following issues have been fixed:
- missing `status`
- WebUI normal display login status

File: adapters/satori/src/ws.ts [hunk 2]
```diff
@@ -47,7 +47,7 @@ export class SatoriAdapter<C extends Context = Context> extends Adapter.WsClient
     // Do not dispatch event from outside adapters.
     let bot = this.bots.find(bot => bot.selfId === selfId && bot.platform === platform)
     if (bot) {
-      bot.update(login)
+      login && bot.update(login)
```
Comment (shigma) [inline adapters/satori/src/ws.ts:50]:
```suggestion
      if (login) bot.update(login)
```

**Review by shigma — APPROVED** [review]

**Pull request merged**